btree/btree.c

```diff
@@ -3148,7 +3173,12 @@ static struct m0_sm_state_descr btree_states[P_NR] = {
 	[P_ACT] = {
 		.sd_flags   = 0,
 		.sd_name    = "P_ACT",
-		.sd_allowed = M0_BITS(P_CLEANUP, P_DONE),
+		.sd_allowed = M0_BITS(P_FREENODE, P_CLEANUP, P_DONE),
+	},
+	[P_FREENODE] = {
+		.sd_flags   = 0,
+		.sd_name    = "P_FREENODE",
+		.sd_allowed = M0_BITS(P_FREENODE, P_CLEANUP, P_DONE),
```

Comment:
P_DONE is not allowed state from P_FREENODE.

Reply:
It is needed , as there is fall throgh from P_FREENODE to P_CLEANUP

Reply:
The fall is from P_FREENODE to P_CLEANUP not P_DONE, right ?

Reply:
Yes but P_CLEANUP will return P_DONE ,So, P_FREENODE  will indirectly return P_DONE